lang/java/avro/src/main/java/org/apache/avro/file/SeekableByteArrayInput.java

```diff
@@ -34,8 +35,12 @@ public long length() throws IOException {
 
   @Override
   public void seek(long p) throws IOException {
-    this.reset();
-    this.skip(p);
+    if (p >= this.count) {
+      throw new EOFException();
+    }
+    if (p >= 0) {
+      this.pos = (int) p;
+    }
```

Comment:
A bit longer this way, but the original would go wrong when `mark()` was ever called.